olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -769,6 +901,74 @@ client::CancellationToken AuthenticationClient::Impl::SignOut(
   });
 }
 
+client::CancellationToken AuthenticationClient::Impl::IntrospectApp(
+    std::string access_token, IntrospectAppCallback callback) {
+  using ResponseType =
+      client::ApiResponse<IntrospectAppResult, client::ApiError>;
+
+  auto introspect_app_task =
+      [=](client::CancellationContext context) -> ResponseType {
+    if (!settings_.network_request_handler) {
+      client::ApiError result({static_cast<int>(http::ErrorCode::IO_ERROR),
+                               "Cannot introspect app while offline"});
+      return std::move(result);
+    }
+
+    client::AuthenticationSettings auth_settings;
+    auth_settings.provider = [&access_token]() { return access_token; };
+
+    client::OlpClientSettings settings;
+    settings.network_request_handler = settings_.network_request_handler;
+    settings.task_scheduler = settings_.task_scheduler;
+    settings.proxy_settings = settings_.network_proxy_settings;
+    settings.authentication_settings = auth_settings;
+
+    client::OlpClient client;
+    client.SetBaseUrl(settings_.token_endpoint_url);
+    client.SetSettings(settings);
+
+    std::multimap<std::string, std::string> query_params;
+    std::multimap<std::string, std::string> header_params;
+    std::multimap<std::string, std::string> forms_params;
```

Comment:
These are input parameters, so if you don't need them then just pass default constructed values instead:

```suggestion
   auto http_result = client.CallApi(kIntrospectAppEndpoint, "GET",
                                      {}, {}, {}, nullptr, content_type, context);
```